pkg/controller/v1alpha1/localmodelnode/controller.go

```diff
@@ -45,13 +53,209 @@ type LocalModelNodeReconciler struct {
+	FSGroup         *int64                                // Can be overwritten by the value in the configmap
+	jobNamespace    string
+	nodeName        = os.Getenv("NODE_NAME") // Name of current node, passed as an env variable via downward API
+	mountPath       = "/mnt/models"          // Volume mount path for models, must be the same as the value in the DaemonSet spec
+)
+
+// Launch a new job or return an existing job
+func (c *LocalModelNodeReconciler) launchJob(ctx context.Context, jobName string, localModelNode *v1alpha1api.LocalModelNode, modelInfo v1alpha1api.LocalModelInfo, claimName string) (*batchv1.Job, error) {
+	container, err := c.getContainerSpecForStorageUri(ctx, modelInfo.SourceModelUri)
+	if err != nil {
+		return nil, err
+	}
+	jobs := c.Clientset.BatchV1().Jobs(jobNamespace)
+
+	job, err := jobs.Get(ctx, jobName, metav1.GetOptions{})
+
+	// In tests, job is an empty struct, using this bool is easier than checking for empty struct
+	jobFound := true
+	if err != nil {
+		if apierr.IsNotFound(err) {
+			jobFound = false
+		} else {
+			c.Log.Error(err, "Failed to get job", "name", jobName)
+			return job, err
+		}
+	}
+
+	container.Name = jobName
+	container.Args = []string{modelInfo.SourceModelUri, mountPath}
+	container.VolumeMounts = []v1.VolumeMount{
+		{
+			MountPath: mountPath,
+			Name:      "kserve-pvc-source",
+			ReadOnly:  false,
+			SubPath:   "models/" + modelInfo.ModelName,
+		},
+	}
+	expectedJob := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName + "dryrun",
+			Namespace: jobNamespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					NodeName:      nodeName,
+					Containers:    []v1.Container{*container},
+					RestartPolicy: v1.RestartPolicyNever,
+					Volumes: []v1.Volume{
+						{
+							Name: "kserve-pvc-source",
+							VolumeSource: v1.VolumeSource{
+								PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+									ClaimName: claimName,
+								},
+							},
+						},
+					},
+					SecurityContext: &v1.PodSecurityContext{
+						FSGroup: FSGroup,
+					},
+				},
+			},
+		},
+	}
+	dryrunJob, err := jobs.Create(ctx, expectedJob, metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}})
+	if err != nil {
+		return nil, err
+	}
+	if job != nil && reflect.DeepEqual(job.Spec.Template.Spec, dryrunJob.Spec.Template.Spec) {
+		return job, nil
+	}
+	if jobFound {
+		bg := metav1.DeletePropagationBackground
+		err = jobs.Delete(ctx, job.Name, metav1.DeleteOptions{
+			PropagationPolicy: &bg,
+		})
+		if err != nil {
+			c.Log.Error(err, "Failed to delete job.", "name", job.Name)
+			return nil, err
+		}
+	}
+
+	if err := controllerutil.SetControllerReference(localModelNode, expectedJob, c.Scheme); err != nil {
+		c.Log.Error(err, "Failed to set controller reference", "name", modelInfo.ModelName)
+		return nil, err
+	}
+	expectedJob.Name = jobName
```

Comment:
it is not necessary to change the job name back as we can use the original name for dry run

Reply:
If the job already exists, I remember dryrun would fail because of the name conflict.